src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -2856,8 +2857,8 @@ private static Value buildArrayObject(Block.Builder builder, List<Value> element
             Value array = builder.op(newArray(J_L_OBJECT_ARRAY, builder.op(constant(INT, elements.size()))));
             for (int i = 0; i < elements.size(); i++) {
                 Value ele = elements.get(i);
-                if (isPrimitive(ele.type())) {
-                    TypeElement wt = getWrapperType(ele.type());
+                if (ele.type() instanceof PrimitiveType pt && pt.box().isPresent()) {
```

Comment:
In this case, returning `null` and using another `instanceof` would have led to better results